instrumentation/micrometer/micrometer-1.5/testing/src/main/java/io/opentelemetry/instrumentation/micrometer/v1_5/AbstractTimerTest.java

```diff
@@ -167,17 +170,22 @@ void testMicrometerTimerWithCustomBuckets() {
                     metric ->
                         assertThat(metric)
                             .hasDescription("This is a test timer")
-                            .hasUnit("ms")
+                            .hasUnit("s")
                             .hasHistogramSatisfying(
                                 histogram ->
                                     histogram.hasPointsSatisfying(
                                         point ->
                                             point
-                                                .hasSum(555500)
+                                                .hasSum(555.5)
                                                 .hasCount(4)
                                                 .hasAttributes(attributeEntry("tag", "value"))
-                                                .hasBucketBoundaries(
-                                                    1_000, 10_000, 100_000, 1_000_000)
+                                                .satisfies(hasBucketBoundaries(1, 10, 100, 1_000))
                                                 .hasBucketCounts(1, 1, 1, 1, 0)))));
   }
+
+  private static ThrowingConsumer<HistogramPointData> hasBucketBoundaries(double... buckets) {
+    return pointData ->
+        assertThat(pointData.getBoundaries().stream().mapToDouble(d -> d).toArray())
+            .contains(buckets, within(0.00001));
+  }
```

Comment:
@jack-berg WDYT about adding a `hasBucketBoundaries(double[], Offset)` variant to the assertion class? (or, to all assertion methods on doubles maybe)

Reply:
Why do you want to include an offset versus just using `hasBucketBoundaries(double... boundaries)`? (Sorry if I'm not seeing something obvious 🙂)

Reply:
Because unfortunately to some double shenanigans the last bucket ends up being 1000.000000000000001 🙈

Reply:
Ah I see. `Offset` is a margin of error in this case. Sorry, just waking up. Yeah I think it would make sense to add that.